src/components/deployment/entity.ts

```diff
@@ -0,0 +1,128 @@
+import { Client } from 'pg';
+import { v4 as uuid } from 'uuid';
+
+import { IDeployment, IEntity, IQuery } from './types';
+
+const deploymentsTable = 'deployments';
+
+interface IBuiltQuery {
+  readonly query: ReadonlyArray<string>;
+  readonly values: ReadonlyArray<any>;
+}
+
+interface IBuiltQueryParams {
+  readonly key: string;
+  readonly operator?: string;
+  readonly value: any;
+}
+
+function where(startAt: number, ...filters: ReadonlyArray<IBuiltQueryParams>): IBuiltQuery {
```

Comment:
I'm not really a fan of this abstraction and think that the plain SQL would be clearer and would not result in more code overall